mlos_bench/mlos_bench/tests/environments/local_env_test.py

```diff
@@ -66,3 +68,132 @@ def test_local_env(tunable_groups: TunableGroups) -> None:
+    )
+
+    with local_env as env_context:
+
+        assert env_context.setup(tunable_groups)
+        (status, _data) = env_context.run()
+        assert status.is_succeeded()
+
+        (status, telemetry) = env_context.status()
+        assert status.is_good()
+        assert telemetry == [
+            (ts1, "cpu_load", 0.65),
+            (ts1, "mem_usage", 10240.0),
+            (ts2, "cpu_load", 0.8),
+            (ts2, "mem_usage", 20480.0),
+        ]
+
+
+def test_local_env_wrong_format(tunable_groups: TunableGroups) -> None:
+    """
+    Fail if the results are not in the expected format.
+    """
+    local_env = LocalEnv(
+        name="Test Local Env",
+        config={
+            "run": [
+                # No header
+                "echo 'latency,10' > output.csv",
+                "echo 'throughput,66' >> output.csv",
+                "echo 'score,0.9' >> output.csv",
+            ],
+            "read_results_file": "output.csv",
+        },
+        service=LocalExecService(parent=ConfigPersistenceService()),
+    )
+    with local_env as env_context:
+        assert env_context.setup(tunable_groups)
+        (status, data) = env_context.run()
+        assert status.is_failed()
+        assert data is None
+
+
+def test_local_env_wide(tunable_groups: TunableGroups) -> None:
+    """
+    Produce benchmark data in wide format and read it.
+    """
+    local_env = LocalEnv(
+        name="Test Local Env",
+        config={
+            "run": [
+                "echo 'latency,throughput,score' > output.csv",
+                "echo '10,66,0.9' >> output.csv",
+            ],
+            "read_results_file": "output.csv",
+        },
+        tunables=tunable_groups,
+        service=LocalExecService(parent=ConfigPersistenceService()),
+    )
+    with local_env as env_context:
+        assert env_context.setup(tunable_groups)
+        (status, data) = env_context.run()
+        assert status.is_succeeded()
+        assert data == {
+            "latency": 10.0,
+            "throughput": 66.0,
+            "score": 0.9,
+        }
+
+
+def test_local_env_vars(tunable_groups: TunableGroups) -> None:
+    """
+    Check that LocalEnv can set shell environment variables.
+    """
+    local_env = LocalEnv(
+        name="Test Local Env",
+        config={
+            "const_args": {
+                "const_arg": 111,  # Passed into "shell_env_params"
+                "other_arg": 222,  # NOT passed into "shell_env_params"
+            },
+            "tunable_params": ["kernel"],
+            "shell_env_params": [
+                "const_arg",                # From "const_arg"
+                "kernel_sched_latency_ns",  # From "tunable_params"
+            ],
+            "run": [
+                "echo const_arg,other_arg,unknown_arg,kernel_sched_latency_ns > output.csv",
+                "echo $const_arg,$other_arg,$unknown_arg,$kernel_sched_latency_ns >> output.csv",
```

Comment:
Would be good to also test this in a composite nested env.
That's where I've run into issues previously.

Reply:
yes, I plan to add many more tests for the composite environment

Reply:
I assume that's for a future PR then?

Reply:
yes